packages/ethers/src/provider-adapter.ts

```diff
@@ -13,30 +12,23 @@ import {
 import { defineReadOnly } from "@ethersproject/properties";
 import { JsonRpcProvider } from "@ethersproject/providers";
 import { AccountSigner } from "./account-signer.js";
-
-export type EthersProviderAdapterOpts = { entryPointAddress?: Address } & (
-  | {
-      rpcProvider: string | PublicErc4337Client<HttpTransport>;
-      chainId: number;
-    }
-  | {
-      accountProvider: SmartAccountProvider<HttpTransport>;
-    }
-);
+import { EthersProviderAdapterOptsSchema } from "./schema.js";
+import type { EthersProviderAdapterOpts } from "./types.js";
 
 /** Lightweight Adapter for SmartAccountProvider to enable Signer Creation */
 export class EthersProviderAdapter extends JsonRpcProvider {
   readonly accountProvider: SmartAccountProvider<HttpTransport>;
 
   constructor(opts: EthersProviderAdapterOpts) {
+    EthersProviderAdapterOptsSchema.parse(opts);
+
     super();
     if ("accountProvider" in opts) {
       this.accountProvider = opts.accountProvider;
     } else {
       const chain = getChain(opts.chainId);
       this.accountProvider = new SmartAccountProvider({
         rpcProvider: opts.rpcProvider,
-        entryPointAddress: opts.entryPointAddress,
```

Comment:
why is this removed? this was here to support optional entry point override for ethers provider adapter?

Reply:
oh my bad, added back